src/Authenticator/ChangePasswordHandler.php

```diff
@@ -0,0 +1,149 @@
+    use LoginHandlerTrait;
+
+    private static $url_handlers = [
+        'GET mfa/schema' => 'getSchema', // Provides details about existing registered methods, etc.
+        'GET mfa/login/$Method' => 'startMFACheck', // Initiates login process for $Method
+        'POST mfa/login/$Method' => 'verifyMFACheck', // Verifies login via $Method
+        'GET mfa' => 'mfa', // Renders the MFA Login Page to init the app
+    ];
+
+    private static $allowed_actions = [
+        'changepassword',
+        'mfa',
+        'getSchema',
+        'startMFACheck',
+        'verifyMFACheck',
+    ];
+
+    protected function jsonResponse(array $response, $code = 200)
+    {
+        return HTTPResponse::create(json_encode($response))
+            ->addHeader('Content-Type', 'application/json')
+            ->setStatusCode($code);
+    }
+
+    public function getSchema()
+    {
+        try {
+            $member = $this->getStore()->getMember();
+            $schema = SchemaGenerator::create()->getSchema($member);
+            return $this->jsonResponse(
+                array_merge($schema, [
+                    'endpoints' => [
+                        'login' => $this->Link('mfa/login/{urlSegment}'),
+                        'complete' => $this->Link(),
+                    ],
+                    'shouldRedirect' => false,
+                ])
+            );
+        } catch (Throwable $exception) {
+            // If we don't have a valid member we shouldn't be here...
+            return $this->redirectBack();
+        }
+    }
+
+    public function mfa()
+    {
+        $store = $this->getStore();
+        if (!$store || !$store->getMember()) {
+            return $this->redirectBack();
+        }
+
+        $this->applyRequirements();
+
+        return [
+            'Form' => $this->renderWith($this->getViewerTemplates()),
+            'ClassName' => 'mfa',
+        ];
+    }
+
+    public function startMFACheck(HTTPRequest $request)
+    {
+        $store = $this->getStore();
+        $member = $store->getMember();
+
+        // If we don't have a valid member we shouldn't be here...
+        if (!$member) {
+            return $this->jsonResponse(['message' => 'Forbidden'], 403);
+        }
+
+        // Use the provided trait method for handling login
+        $response = $this->createStartLoginResponse(
+            $store,
+            Injector::inst()->get(MethodRegistry::class)->getMethodByURLSegment($request->param('Method'))
+        );
+
+        // Ensure detail is saved to the store
+        $store->save($request);
+
+        // Respond with our method
+        return $response;
+    }
+
+    public function verifyMFACheck(HTTPRequest $request)
+    {
+        $store = $this->getStore();
+
+        try {
+            $result = $this->verifyLoginRequest($store, $request);
+        } catch (InvalidMethodException $e) {
+            // Invalid method usually means a timeout. A user might be trying to verify before "starting"
```

Comment:
Again - do you think it might be worth logging these exceptions instead of swallowing them?

Reply:
Logging added